### PR TITLE
add a message and fix error when on multishop

### PR DESCRIPTION
**Pull request by pablopolyte**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | add message when on multishop and fix error
| Type?             | bug fix / improvement
| BC breaks?        | no
| Deprecations?     | no

### CODE REVIEW

File: src/Controller/WishlistConfigurationAdminController.php [hunk 1]
```diff
@@ -38,11 +38,11 @@ class WishlistConfigurationAdminController extends FrameworkBundleAdminControlle
     private $cache;
 
     /**
-     * @var int
+     * @var int|null
      */
     private $shopId;
 
-    public function __construct(CacheProvider $cache, int $shopId)
+    public function __construct(CacheProvider $cache, $shopId)
```
Comment (Progi1984) [inline src/Controller/WishlistConfigurationAdminController.php:45]:
```suggestion
    public function __construct(CacheProvider $cache, ?int $shopId)
```

File: views/templates/admin/statistics.html.twig [hunk 2]
```diff
@@ -19,6 +19,14 @@
 {% extends '@PrestaShop/Admin/layout.html.twig' %}
 
 {% block content %}
+  {% if shopId == null %}
```
Comment (Progi1984) [inline views/templates/admin/statistics.html.twig:22]:
```suggestion
  {% if shopId === null %}
```

File: src/Grid/Data/BaseGridDataFactory.php [hunk 3]
```diff
@@ -19,11 +19,11 @@ class BaseGridDataFactory
     protected $calculator;
 
     /**
-     * @var int
+     * @var int|null
      */
     protected $shopId;
 
-    public function __construct(CacheProvider $cache, StatisticsCalculator $calculator, int $shopId)
+    public function __construct(CacheProvider $cache, StatisticsCalculator $calculator, $shopId)
```
Comment (Progi1984) [inline src/Grid/Data/BaseGridDataFactory.php:26]:
```suggestion
    public function __construct(CacheProvider $cache, StatisticsCalculator $calculator, ?int $shopId)
```

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

File: src/Controller/WishlistConfigurationAdminController.php (hunk 1 above)
Comment (pablopolyte) [reply src/Controller/WishlistConfigurationAdminController.php:45]: don't fit php checker 5.6

File: src/Grid/Data/BaseGridDataFactory.php (hunk 3 above)
Comment (pablopolyte) [reply src/Grid/Data/BaseGridDataFactory.php:26]: don't fit php checker 5.6

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @pablopolyte

File: src/Controller/WishlistConfigurationAdminController.php (hunk 1 above)
Comment (pablopolyte) [reply src/Controller/WishlistConfigurationAdminController.php:45]: this cast don't fit the php 5.6 checker

File: src/Grid/Data/BaseGridDataFactory.php (hunk 3 above)
Comment (pablopolyte) [reply src/Grid/Data/BaseGridDataFactory.php:26]: same here

**Pull request merged**